google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/Util.java

```diff
@@ -148,4 +151,45 @@ static Long getGfeLatency(Metadata metadata) {
     }
     return null;
   }
+
+  private static ResponseParams getResponseParams(@Nullable Metadata metadata) {
+    byte[] responseParams;
+    if (metadata != null && (responseParams = metadata.get(Util.LOCATION_METADATA_KEY)) != null) {
+      try {
+        return ResponseParams.parseFrom(responseParams);
+      } catch (InvalidProtocolBufferException e) {
+      }
+    }
+    return null;
+  }
+
+  static void recordMetricsFromMetadata(
+      GrpcResponseMetadata responseMetadata, BigtableTracer tracer, Throwable throwable) {
+    // server-timing metric will be added through GrpcResponseMetadata#onHeaders(Metadata),
+    // so it's not checking trailing metadata here.
+    Metadata metadata = responseMetadata.getMetadata();
+    Long latency = getGfeLatency(metadata);
```

Comment:
`@Nullable`

Reply:
Also group the latency stuff together 

Metadata metadata = responseMetadata.getMetadata();

// Location & Cluster
ResponseParams responseParams ...

// Latency
Long latency = getGfeLatency(metadata);
if (responseParams != null && latency == null) {
      latency = 0L;
    }
tracer.recordGfeMetadata(latency, throwable);